services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/attachment/AttachmentController.kt

```diff
@@ -0,0 +1,234 @@
+    @Autowired private val attachmentService: AttachmentService,
+    @Autowired private val hankeService: HankeService,
+    @Autowired private val permissionService: PermissionService,
+) {
+
+    @GetMapping("liite/{id}")
+    @Operation(
+        summary = "Get attachment metadata by UUID",
+        description = "Return information about an attachment without binary data."
+    )
+    @ApiResponses(
+        value =
+            [
+                ApiResponse(
+                    description = "Get attachment by Id.",
+                    responseCode = "200",
+                    content = [Content(schema = Schema(implementation = AttachmentMetadata::class))]
+                ),
+                ApiResponse(
+                    description = "Attachment not found.",
+                    responseCode = "404",
+                    content =
+                        [
+                            Content(
+                                schema = Schema(implementation = AttachmentNotFoundException::class)
+                            )
+                        ]
+                ),
+            ]
+    )
+    fun getAttachment(@PathVariable id: UUID): AttachmentMetadata {
+        val attachmentMetadata = attachmentService.getMetadata(id)
+        val hankeId = hankeService.getHankeId(attachmentMetadata.hankeTunnus)
+
+        checkAttachmentPermission(hankeId, currentUserId(), VIEW)
+
+        return attachmentMetadata
+    }
+
+    @GetMapping("liite/{id}/content")
+    @Operation(summary = "Download attachment file.", description = "Returns the uploaded file.")
+    @ApiResponses(
+        value =
+            [
+                ApiResponse(
+                    description = "Download content of the attachment file.",
+                    responseCode = "200"
+                ),
+                ApiResponse(
+                    description = "Attachment not found.",
+                    responseCode = "404",
+                    content =
+                        [
+                            Content(
+                                schema = Schema(implementation = AttachmentNotFoundException::class)
+                            )
+                        ]
+                ),
+            ]
+    )
+    fun getAttachmentContent(
+        @PathVariable id: UUID,
+        response: HttpServletResponse
+    ): ResponseEntity<ByteArray> {
+        val attachmentMetadata = attachmentService.getMetadata(id)
+        val hankeId = hankeService.getHankeId(attachmentMetadata.hankeTunnus)
+
+        checkAttachmentPermission(hankeId, currentUserId(), VIEW)
+
+        val metadata = attachmentService.getMetadata(id)
+        val mimeType = URLConnection.guessContentTypeFromName(metadata.fileName)
+
+        val responseHeaders =
+            HttpHeaders().apply {
+                contentType = MediaType.parseMediaType(mimeType)
+                add(CONTENT_DISPOSITION, "attachment; filename=${metadata.fileName}")
+            }
```

Comment:
What's wrong with just plain old
```suggestion
        val headers = HttpHeaders()
        headers.contentType = MediaType.parseMediaType(mimeType)
        headers.add(CONTENT_DISPOSITION, "attachment; filename=${metadata.fileName}")
```

Reply:
Fixed.